addok-psql-store: psycopg2.pool.PoolError: connection pool exhausted

```
[2017-04-20 15:31:08 +0200] [30171] [ERROR] Error handling request /search/?q=44+Rue+de+l%27Amiral+Bruix+Boulogne-sur-Mer&limit=1
Traceback (most recent call last):
  File "/home/ybon/.virtualenvs/addok/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/home/ybon/.virtualenvs/addok/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 176, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/home/ybon/.virtualenvs/addok/lib/python3.6/site-packages/falcon-1.1.0-py3.6.egg/falcon/api.py", line 209, in __call__
    responder(req, resp, **params)
  File "/home/ybon/Code/py/addok/addok/http/base.py", line 134, in on_get
    lat=lat, lon=lon, **filters)
  File "/home/ybon/Code/py/addok/addok/core.py", line 327, in search
    return helper(query, lat=lat, lon=lon, **filters)
  File "/home/ybon/Code/py/addok/addok/core.py", line 169, in __call__
    return self.render()
  File "/home/ybon/Code/py/addok/addok/core.py", line 183, in render
    self.convert()
  File "/home/ybon/Code/py/addok/addok/core.py", line 227, in convert
    for _id, doc in documents:
  File "/home/ybon/Code/py/addok/addok/ds.py", line 83, in get_documents
    for id_, blob in DS.fetch(*keys):
  File "/home/ybon/Code/py/addok-psql-store/addok_psql_store/__init__.py", line 28, in fetch
    with self.pool.getconn() as conn, conn.cursor() as curs:
  File "/home/ybon/.virtualenvs/addok/lib/python3.6/site-packages/psycopg2/pool.py", line 91, in _getconn
    raise PoolError("connection pool exhausted")
psycopg2.pool.PoolError: connection pool exhausted
```

When running geocoder-tester against a local addok instance with gunicorn.

--- a/addok_psql_store/__init__.py
+++ b/addok_psql_store/__init__.py
@@ -1,3 +1,5 @@
+import os
+
 from psycopg2 import pool
 from psycopg2.extras import execute_values
 
@@ -16,16 +18,19 @@ class PSQLStore:
         CREATE UNIQUE INDEX IF NOT EXISTS
             {PG_TABLE}_key_idx ON {PG_TABLE} (key)
         '''.format(**config)
-        with self.pool.getconn() as conn, conn.cursor() as curs:
+        with self.getconn() as conn, conn.cursor() as curs:
             curs.execute(create_table_query)
             curs.execute(create_index_query)
+
+    def getconn(self):
+        return self.pool.getconn(key=os.getpid())
 
     def fetch(self, *keys):
         # Using ANY results in valid SQL if `keys` is empty.
         select_query = '''
         SELECT key, data FROM {PG_TABLE} WHERE key=ANY(%s)
         '''.format(**config)
-        with self.pool.getconn() as conn, conn.cursor() as curs:
+        with self.getconn() as conn, conn.cursor() as curs:
             curs.execute(select_query, ([key.decode() for key in keys],))
             for key, data in curs.fetchall():
                 yield key.encode(), data
@@ -43,21 +48,21 @@ class PSQLStore:
         INSERT INTO {PG_TABLE} (key, data) VALUES %s
             ON CONFLICT DO NOTHING
         '''.format(**config)
-        with self.pool.getconn() as conn, conn.cursor() as curs:
+        with self.getconn() as conn, conn.cursor() as curs:
             execute_values(curs, insert_into_query, docs)
 
     def remove(self, *keys):
         delete_from_query = '''
         DELETE FROM {PG_TABLE} WHERE key=%s
         '''.format(**config)
-        with self.pool.getconn() as conn, conn.cursor() as curs:
+        with self.getconn() as conn, conn.cursor() as curs:
             curs.executemany(delete_from_query, (keys, ))
 
     def flushdb(self):
         drop_table_query = '''
         DROP TABLE IF EXISTS {PG_TABLE}
         '''.format(**config)
-        with self.pool.getconn() as conn, conn.cursor() as curs:
+        with self.getconn() as conn, conn.cursor() as curs:
             curs.execute(drop_table_query)
 
 

